### ISSUE TITLE
Error updating Rowan using Rowan from v0.3.1-alpha to facf07b (current candidate)

### ISSUE THREAD

**Issue by dalehenrich** [dalehenrich/Rowan#260]
Need to come up with a simpler test case for this and imlement the son of a gun:
```
a UserDefinedError occurred (error 2318), reason:halt, not yet implemented
---
UserDefinedError (AbstractException) >> _signalWith: @5 line 25
UserDefinedError (AbstractException) >> signal @2 line 47
RwGsClassVersioningSymbolDictPatch (Object) >> error: @6 line 7
[] in  ExecBlock1 (RwGsClassVersioningSymbolDictPatch) >> _updateNewClassVersionPatchesForClass:in:patchSet: @15 line 32
[] in  ExecBlock1 (Set) >> do: @2 line 11
[] in  ExecBlock2 (KeyValueDictionary) >> keysDo: @2 line 7
KeyValueDictionary >> keysAndValuesDo: @18 line 15
KeyValueDictionary >> keysDo: @2 line 6
Set >> do: @3 line 11
[] in  ExecBlock1 (RwGsPatchSet) >> allPatchesAffectingLiveMethodsDo:deletedMethodsDo: @2 line 7
Array (Collection) >> do: @5 line 10
RwGsPatchSet >> allPatchesAffectingLiveMethodsDo:deletedMethodsDo: @2 line 7
RwGsClassVersioningSymbolDictPatch >> _updateNewClassVersionPatchesForClass:in:patchSet: @15 line 23
RwGsClassVersioningSymbolDictPatch >> updateNewClassVersionPatchesForExtensionsIn:patchSet: @6 line 7
[] in  ExecBlock1 (RwGsImage class) >> applyModification:visitorClass:instanceMigrator: @3 line 17
[] in  ExecBlock1 (Set) >> do: @2 line 11
[] in  ExecBlock2 (KeyValueDictionary) >> keysDo: @2 line 7
KeyValueDictionary >> keysAndValuesDo: @18 line 15
KeyValueDictionary >> keysDo: @2 line 6
Set >> do: @3 line 11
RwGsImage class >> applyModification:visitorClass:instanceMigrator: @14 line 13
RwGsImage class >> applyModification:instanceMigrator: @7 line 8
RwPrjLoadTool >> _loadProjectSetDefinition:instanceMigrator: @17 line 19
[] in  ExecBlock0 (RwPrjLoadTool) >> _doProjectSetLoad:instanceMigrator:originalProjectSet:processedClassNames: @2 line 4
ExecBlock0 (ExecBlock) >> on:do: @3 line 42
RwPrjLoadTool >> _doProjectSetLoad:instanceMigrator:originalProjectSet:processedClassNames: @2 line 6
RwPrjLoadTool >> loadProjectNamed:instanceMigrator: @7 line 9
RwPrjLoadTool >> loadProjectNamed: @4 line 5
[] in  ExecBlock0 (RowanProjectService) >> loadProjectNamed: @4 line 3
ExecBlock0 (ExecBlock) >> on:do: @3 line 42
RowanProjectService >> loadProjectNamed: @2 line 4
GsNMethod class >> _gsReturnToC @1 line 1
```

**Comment by dalehenrich** [comment]
The original stack occurs when methods are actually modified are modified, however, this stack occurs when the the extension methods are not modified:
```
a LookupError occurred (error 2021), reason:rtErrKeyNotFound, A reference using the non-existent key 'Simple NewVersionD 2' was made into the dictionary aDictionary( 'Simple NewVersionD 1'->aRwProjectModification)
---
LookupError (AbstractException) >> _signalWith: @5 line 25
LookupError (AbstractException) >> signal @2 line 47
Dictionary (Object) >> _error:args: @15 line 11
Dictionary (AbstractDictionary) >> _errorKeyNotFound: @2 line 5
Dictionary >> at: @5 line 8
RwGsClassVersioningSymbolDictPatch >> _createMethodExtensionModificationForNewVersionOfClassNamed:isMeta:loadedMethod:projectSetModification: @10 line 8
[] in  ExecBlock1 (RwGsClassVersioningSymbolDictPatch) >> _updateNewClassVersionPatchesForClass:in:patchSet: @2 line 47
Dictionary >> valuesDo: @8 line 13
[] in  ExecBlock1 (RwGsClassVersioningSymbolDictPatch) >> _updateNewClassVersionPatchesForClass:in:patchSet: @3 line 45
Dictionary >> valuesDo: @8 line 13
RwGsClassVersioningSymbolDictPatch >> _updateNewClassVersionPatchesForClass:in:patchSet: @16 line 43
RwGsClassVersioningSymbolDictPatch >> updateNewClassVersionPatchesForExtensionsIn:patchSet: @6 line 7
[] in  ExecBlock1 (RwGsImage class) >> applyModification_254:visitorClass:instanceMigrator: @3 line 17
[] in  ExecBlock1 (Set) >> do: @2 line 11
[] in  ExecBlock2 (KeyValueDictionary) >> keysDo: @2 line 7
KeyValueDictionary >> keysAndValuesDo: @23 line 20
KeyValueDictionary >> keysDo: @2 line 6
Set >> do: @3 line 11
RwGsTestImage class (RwGsImage class) >> applyModification_254:visitorClass:instanceMigrator: @14 line 13
RwGsTestImage class (RwGsImage class) >> applyModification_254:instanceMigrator: @8 line 12
RwPrjLoadTool >> _loadProjectSetDefinition_254:instanceMigrator: @14 line 14
[] in  ExecBlock0 (RwPrjLoadTool) >> _doProjectSetLoad_254:instanceMigrator:originalProjectSet:processedClassNames: @2 line 4
ExecBlock0 (ExecBlock) >> on:do: @3 line 42
RwPrjLoadTool >> _doProjectSetLoad_254:instanceMigrator:originalProjectSet:processedClassNames: @2 line 6
RwPrjLoadTool >> loadProjectSetDefinition:instanceMigrator: @3 line 7
RwPrjLoadTool >> loadProjectSetDefinition: @4 line 4
RwPrjBrowserTool >> projectsNamed:updateDefinition: @7 line 11
RwBrowserToolApiTest >> testNewClassVersion_260_unchanged_extension_method_protocol @59 line 102
RwBrowserToolApiTest (TestCase) >> performTest @3 line 2
[] in  ExecBlock0 (TestCase) >> runCase @3 line 5
ExecBlock0 (ExecBlock) >> ensure: @2 line 12
RwBrowserToolApiTest (TestCase) >> runCase @4 line 6
[] in  ExecBlock0 (TestCase) >> debug @4 line 2
ExecBlock0 (ExecBlock) >> ensure: @2 line 12
RwBrowserToolApiTest (TestCase) >> debug @2 line 3
RwBrowserToolApiTest class (TestCase class) >> debug: @3 line 3
[] in  ExecBlock1 (JadeServer) >> sbRunMethodTests: @3 line 6
Array (Collection) >> do: @5 line 10
JadeServer64bit32 (JadeServer) >> sbRunMethodTests: @8 line 6
GsNMethod class >> _gsReturnToC @1 line 1
```
Showed up when writing tests to reproduce original stack

**Comment by dalehenrich** [comment]
The basic root cause of this bug is to create a new version of a class that has extension methods ... a not uncommon use case ... needs to be fixed for v1.0.0
